### ISSUE TITLE
[luajit] build failure

### ISSUE THREAD

**Issue by bugbuka** [microsoft/vcpkg#11809]
**Host Environment**
- OS: [macos 10.15.4]
- Compiler: revision

**To Reproduce**
`vcpkg install luajit --triplet x64-osx`

**Failure logs** 
```
Kais-MacBook-Pro:minetest tom$ vcpkg install luajit --triplet x64-osx
Computing installation plan...
The following packages will be built and installed:
    luajit[core]:x64-osx
Starting package 1/1: luajit:x64-osx
Building package luajit[core]:x64-osx...
-- Using cached /Users/tom/github-cpp/vcpkg/downloads/LuaJIT-LuaJIT-v2.0.5.tar.gz
-- Using source at /Users/tom/github-cpp/vcpkg/buildtrees/luajit/src/v2.0.5-045e1daa09
-- Building x64-osx-dbg
CMake Error at scripts/cmake/vcpkg_execute_required_process_repeat.cmake:23 (message):
    Command failed: /Users/tom/github-cpp/vcpkg/buildtrees/luajit/src/v2.0.5-045e1daa09/src/msvcbuild.bat /Users/tom/github-cpp/vcpkg/buildtrees/luajit/src/v2.0.5-045e1daa09/src dynamic debug static
    Working Directory: /Users/tom/github-cpp/vcpkg/buildtrees/luajit/x64-osx-dbg
    See logs for more information:
      /Users/tom/github-cpp/vcpkg/buildtrees/luajit\build-x64-osx-dbg-out.log
      /Users/tom/github-cpp/vcpkg/buildtrees/luajit\build-x64-osx-dbg-err.log

Call Stack (most recent call first):
  ports/luajit/portfile.cmake:25 (vcpkg_execute_required_process_repeat)
  scripts/ports.cmake:90 (include)


Error: Building package luajit:x64-osx failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: luajit:x64-osx
  Vcpkg version: 2020.02.04-unknownhash

Additionally, attach any relevant sections from the log files above.
```
**Additional context**
Add any other context about the problem here, such as what you have already tried to resolve the issue.

**Comment by timautry** [comment]
@bugbuka Thank you for posting this issue.
Please upload/attach the following files to this issue:
`/Users/tom/github-cpp/vcpkg/buildtrees/luajit\build-x64-osx-dbg-out.log`
`/Users/tom/github-cpp/vcpkg/buildtrees/luajit\build-x64-osx-dbg-err.log`

This will allow someone more familiar with macosx to assist with this issue.

**Comment by NancyLi1013** [comment]
`luajit` currently only supports Windows platform. Since it depends on Visual Studio to build.
https://github.com/microsoft/vcpkg/blob/master/ports/luajit/portfile.cmake#L27

I will add the unsupported messages for `luajit`.

**Comment by DailyShana** [comment]
luajit uses GNU Make to build on POSIX platforms
https://github.com/LuaJIT/LuaJIT/blob/v2.1/Makefile

**Comment by NancyLi1013** [comment]
@DailyShana 
Thanks for your above info. I will try to build on Linux and verify if it can work.

**Comment by FrankXie05** [comment]
Fixed by #24584
